edb/pgsql/compiler/relgen.py

```diff
@@ -492,6 +492,32 @@ def set_as_subquery(
     return wrapper
 
 
+def can_omit_optional_wrapper(
+        ir_set: irast.Set, new_scope: irast.ScopeTreeNode, *,
+        ctx: context.CompilerContextLevel) -> bool:
+    """Determine whether it is safe to omit the optional wrapper.
+
+    Doing so is safe when the expression is guarenteed to result in
+    a NULL and not an empty set.
+
+    Currently the only such case implement is a path `foo.bar` where foo
+    is visible and bar is a single non-computed property, which we know
+    will be stored as NULL in the database.
+    """
+    if (
```

Comment:
Why not just `return` the condition?

Reply:
I had in my head that there would be a chain of conditions... but then there was just one :P